diag_manager/include/fms_diag_fieldbuff_update.fh

```diff
@@ -0,0 +1,1371 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+ !> @brief This code will be used by the preprocessor to generate an implementation
+ !! of the module procedure for the fieldbuff_update interface. The
+  !! generated function is a wrapper calling 4D field/5D buffer version of the same.
+FUNCTION FMS_DIAG_FBU_3D_PNAME_(ofield_cfg, ofield_index_cfg, field_d, sample, &
+  & ofb, ofc, bbounds, count_0d, num_elements, mask, weight1, missvalue,  &
+  & field_num_threads, field_active_omp_level, issued_mask_ignore_warning, &
+  & l_start, l_end, err_msg,  err_msg_local ) result( succeded )
+    TYPE(fmsDiagOutfield_type), INTENT(in):: ofield_cfg  !< The fmsDiagOutfield_type object,
+                                                         !! where "cfg" is short for configuration
+    TYPE(fmsDiagOutfieldIndex_type) , INTENT(in) :: ofield_index_cfg !<The fmsDiagOutfieldIndex_type object,
+                                                                !! where "cfg" is short for configuration
+    FMS_DIAG_FBU_DATA_TYPE_ , CONTIGUOUS, DIMENSION(:,:,:), INTENT(in), target :: field_d !< The input field data
```

Comment:
For example, from the comment made for the inc file below, this would be real( FMS_DIAG_FBU_DATA_KIND_)

Reply:
I think `TYPE_` is applicable here because this can/will support integers

Reply:
Integers can be supported with KIND_also.  Also, in that case, KIND_ may be less  confusing since the users can distinguish which variables are integers and which are reals